nautobot/core/views/utils.py

```diff
@@ -516,3 +528,103 @@ def sample_line(line):
     # END Nautobot-specific logic
 
     return "".join(output).encode("utf-8")
+
+
+def get_bulk_queryset_from_view(user, model, is_all, filter_query_params, pk_list, saved_view_id="", log=None):
+    """
+    Return a queryset for bulk operations based on the provided parameters.
+
+    Args:
+        user: The user performing the bulk operation.
+        model: The model class on which the bulk operation is being performed.
+        is_all: Boolean indicating whether the operation applies to pk_list or not.
+        filter_query_params: A request.GET or dictionary of filter parameters to apply to the queryset.
+        pk_list: A list of primary keys to include, when not using a filter.
+        saved_view_id: (Optional) UUID of a saved view to apply additional filters from.
+
+    Returns:
+        A Django queryset representing the objects to be affected by the bulk operation.
+
+    Notes:
+        Start
+        ├── !is_all and pk_list: Return queryset filtered by pk_list
+        ├── !is_all and !pk_list: Return empty queryset
+        ├── is_all and !saved_view_id and ~filter_query_params: Return all objects
+        ├── is_all and filter_query_params: Return queryset filtered by filter_query_params
+        ├── is_all and saved_view_id: Return queryset filtered by saved_view_filter_params
+        └── else: Return empty queryset
+    """
+    # We do this on purpose, as taking from the view will not work once set to the job. It is better
+    # to be consistent from view -> to confirmation view -> job, then to be more correct for edge case views
+    view_class = get_view_for_model(model, view_type="List")
+    view_instance = view_class()
+    request = RequestFactory().get("/")
+    view_instance.request = request
+    view_instance.request.user = user
+    view_instance.action = "bulk_edit"  # or "bulk_delete", doesn't matter here
+    queryset = view_instance.get_queryset().restrict(user, "change")
+
+    if not log:
+        log = logger
+
+    # We do this on purpose, as taking from the view will not work once set to the job. It is better
+    # to be consistent from view -> to confirmation view -> job, then to be more correct for edge case views
+    filterset_class = get_filterset_for_model(model)
+
+    if isinstance(filter_query_params, QueryDict):
+        filterset_class = lookup.get_filterset_for_model(model)
+        if filterset_class:
+            filter_query_params = normalize_querydict(filter_query_params, filterset=filterset_class())
+        else:
+            filter_query_params = {}
+
+    # The form actually sends the pks and the "all" parameter, so seeing pk_list by itself is not
+    # sufficient to determine if we are filtering by pk_list or by all. We need to see is_all=False.
+    if not is_all and pk_list:
+        log.debug(f"Filtering by PKs: {pk_list}")
```

Comment:
This could easily be a very long list of PKs (consider `page_size=1000`), so perhaps:

```suggestion
        log.debug(f"Filtering by PK list")
```